vms/platformvm/block/builder/builder.go

```diff
@@ -270,59 +283,74 @@ func (b *builder) PackAllBlockTxs() ([]*txs.Tx, error) {
 		return nil, fmt.Errorf("could not calculate next staker change time: %w", err)
 	}
 
+	recommendedPChainHeight, err := b.txExecutorBackend.Ctx.ValidatorState.GetMinimumHeight(context.TODO())
+	if err != nil {
+		return nil, err
+	}
+
 	if !b.txExecutorBackend.Config.UpgradeConfig.IsEtnaActivated(timestamp) {
 		return packDurangoBlockTxs(
+			context.TODO(),
```

Comment:
> Is the intention to pass an actual context at some point

Yes

> to limit the scope of execution for a given operation?

The intention is to improve tracing support rather than aborting execution.

> A quick survey suggest that current context usage is a mix of TODO and Background.

Every (real) context starts as a `context.Background()` at some point, additional information gets added to it (like tracing information).